gitlab/v4/objects/commits.py

```diff
@@ -17,6 +17,10 @@
 
 class ProjectCommit(RESTObject):
     _short_print_attr = "title"
+
+    comments: "ProjectCommitCommentManager"
+    discussions: ProjectCommitDiscussionManager
+    statuses: "ProjectCommitStatusManager"
```

Comment:
I see you mix plain classes and strings here, should this be consistent everywhere or is there a specific reason for this?

Reply:
It is because of forward references. The class hasn't been declared yet so get an error.

Could probably work around it by using the Python 3.7 or later feature:

```
from future import __annotations__
```

But we currently support Python 3.6 which is EOL at the end of this year (2021).
https://www.python.org/dev/peps/pep-0494/#lifespan

Reply:
I see, thanks! Maybe we could conditionally add the backport, https://pypi.org/project/future-annotations/#files for folks on 3.6 to the install requires, it's a 6kB package with a single module. It would clean up _create_managers a bit as well.

Reply:
My preference would be to leave it as is. Using strings or proper class names are both valid ways to do it. It is a more of a style issue.

Now when we drop support for Python 3.6, then sure we could add the `from future import __annotations__` though what that does is basically make them all be stored as strings.

There are some disadvantages to doing that as explained in: https://www.python.org/dev/peps/pep-0563/#resolving-type-hints-at-runtime  I am not sure I understand it enough to say which way is best.

Reply:
Ok, I was just thinking because it's a tiny package that would only get installed and add `from future import __annotations__` for 3.6, would probably help us clean this up:
```
            if isinstance(annotation, type):
                cls_name = annotation.__name__
            else:
                cls_name = annotation
```

But true we can revisit this later :)